### PR TITLE
Refactor lets split to use split common code

### PR DESCRIPTION
**Pull request by zvecr**
I've followed what has been done previously to other boards (#3639).

Cherry picked out the changes from my [keymap branch](/zvecr/qmk_firmware/tree/feature/ortho_4x12-keymap) to hopefully make the changes a lot more isolated.

A quick skim of the readme seems like it stands true, but let me know if anything else is required.

### CODE REVIEW

File: keyboards/lets_split/rules.mk [hunk 1]
```diff
@@ -1,11 +1,4 @@
-SRC += matrix.c \
-	   i2c.c \
-	   split_util.c \
-	   serial.c \
-	   ssd1306.c
```
Comment (drashna) [inline keyboards/lets_split/rules.mk]: Could you add `SRC += ssd1306.c` to the OLED_test keymap's rules.mk? 

That should fix it's error.

**Comment by drashna** [comment]
Also, Travis CI errors are caused by @agbell's keymap (tap dance issue, needs to be explicitly enabled in the keymap's rules.mk), the OLED_sample keymap (because the ssd1306.c file isn't properly included), @PyrooL's keymap since it's include is incorrect (it's lowercase), and @drashna (that guy, ugh).

They can (mostly) be ignored.

**Comment by drashna** [comment]
Sorry, nope, fixes aren't required (except for the SSD1306 file for the OLED_sample keymap).  I posted mostly to tag people to clean up their own keymaps.

**Comment by zvecr** [comment]
My mentioning of the other PR was for the other people, in case they want to cherry pick the changes.

File: keyboards/lets_split/rules.mk (hunk 1 above)
Comment (zvecr) [reply keyboards/lets_split/rules.mk]: Done, builds locally but produces a `The firmware is too large!` error. Given the name of the keymap, OLED_sample, would it be worth disabling some features to reduce the size?

**Comment by drashna** [comment]
@PyrooL Well, really Travis CI caught it.  And we've been trying to be a bit more proactive about flagging "problem keymaps", so they can be fixed.  It's an issue that's come up several times recently.

File: keyboards/lets_split/rules.mk (hunk 1 above)
Comment (drashna) [reply keyboards/lets_split/rules.mk]: Yeah. Start with mousekeys, though.

Comment (zvecr) [reply keyboards/lets_split/rules.mk]: All good with just the mouse keys disabled.

**Comment by drashna** [comment]
Looks good! Thanks!

**Pull request merged**